File: AkitaModel/Flagging/ResultEntry.cs
﻿namespace Skyware.Lis.AkitaModel.Flagging;

public class ResultEntry
{

    /// <summary>
    /// Allowed value of the result.
    /// </summary>
    public string Value { get; set; }

    /// <summary>
    /// Corresponding flag for this value.
    /// </summary>
    public FlagLevels FlagLevel { get; set; } = FlagLevels.NONE;

}

File: AkitaMotelTS/ResultEntry.ts
import { FlagLevels } from "./FlagLevels";

/**
 * Allowed result entry.
 */
export class ResultEntry {
  /**
   * Allowed value of the result.
   */
  value: string;

  /**
   * Corresponding flag for this value.
   */
  flagLevel: FlagLevels = FlagLevels.NONE;

  /**
   * Constructor to initialize the ResultEntry class.
   * @param value - Allowed value of the result.
   * @param flagLevel - Corresponding flag level for the value (default is NONE).
   */
  constructor(value: string, flagLevel: FlagLevels = FlagLevels.NONE) {
    this.value = value;
    this.flagLevel = flagLevel;
  }
}
